File: Team Homies/Homies/Backend/models/transaction.js
const Mongoose = require('mongoose');

const TransactionSchema = new Mongoose.Schema({

    transaction_id: {
        type: String,
        required: true,
        unique: true
    },
    sender_id: {
        type: String,
        required: true
    },
    receiver_id: {
        type: String,
        required: true                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
    },
    amount: {
        type: Number,
        required: true
    },
    timestamp: {
        type: String,
        required: true
    }

})

module.exports = Mongoose.model('Transaction', TransactionSchema);

File: Team Homies/Homies/Android-Application/Dhanrakshak2/app/src/main/java/com/example/dhanrakshak/data/model/entities/Transaction.kt
package com.example.dhanrakshak.data.model.entities

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "Transactions")
data class Transaction(
    @PrimaryKey
    val transactionId: String,
    val senderId: String,
    val receiverId: String,
    val senderName: String,
    val amount: Long,
    val timeStamp: Long,
)

class InvalidTransactionException(message: String): Exception(message)